streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -136,20 +136,13 @@ public void flush() {
         checkForException();
```

Comment:
@mjsax there are lots of places in this file where we check for exceptions and throw an exception, sometimes at unexpected times. There is a JIRA open for this at https://issues.apache.org/jira/browse/KAFKA-5006. With EoS these kind of exceptions probably don't make sense to be thrown this way. Also with EoS, I guess we'll need to rollback the transaction. My actual question:
- does it make sense to fix that JIRA as part of this PR?